result4k/core/src/main/kotlin/dev/forkhandles/result4k/iterables.kt

```diff
@@ -18,3 +18,42 @@ fun <T, E> Iterable<Result<T, E>>.partition(): Pair<List<T>, List<E>> {
     }
     return Pair(oks, errs)
 }
+
+// Traverse family of functions
+fun <T, Tʹ, E> Iterable<T>.foldResult(
+    initial: Result<Tʹ, E>,
+    operation: (acc: Tʹ, T) -> Result<Tʹ, E>
+): Result<Tʹ, E> =
+    fold(initial) { acc, el -> acc.flatMap { accVal -> operation(accVal, el) } }
+
+fun <T, Tʹ, E> Iterable<T>.traverse(f: (T) -> Result<Tʹ, E>): Result<List<Tʹ>, E> =
+    foldResult(Success(mutableListOf())) { acc, e ->
+        f(e).map { acc.add(it); acc }
```

Comment:
Not keen on mixing imperative and declarative style here.  I find it confusing that a fold is actually mutating a parameter instead of transforming it.

Reply:
I agree on the principle, but using it with a list of 100k elements or so makes an huge difference in performance. So much that it makes traverse unusable in prod. I don't know of other solutions.
Also since all lists are actually mutable in Kotlin, I don;t see a big problem in modifying it and then returning it as readonly List.